libnetdata/socket/security.c

```diff
@@ -357,31 +357,22 @@ int security_test_certificate(SSL *ssl) {
  * @return It returns 0 on success and -1 otherwise.
  */
 int ssl_security_location_for_context(SSL_CTX *ctx, char *file, char *path) {
-    struct stat statbuf;
-    if (stat(file, &statbuf)) {
-        info("Netdata does not have the parent's SSL certificate, so it will use the default OpenSSL configuration to validate certificates!");
-        return 0;
-    }
-
-    ERR_clear_error();
-    u_long err;
-    char buf[256];
-    if(!SSL_CTX_load_verify_locations(ctx, file, path)) {
-        goto slfc;
+    int load_custom = 1, load_default = 1;
+    if (file || path) {
+        if(!SSL_CTX_load_verify_locations(ctx, file, path)) {
```

Comment:
this probably will fail if `file` doesn't exist but `dir` is a valid dir that exists

Reply:
Well, when the user edits the paths in stream.conf, I would assume they put proper dirs/files there. We don't mess with that, if they fail we print a message (and still use default to do the best we can).

Reply:
ok, thich is not a problem if we NULL the default.